chia/wallet/wallet_state_manager.py

```diff
@@ -742,12 +742,8 @@ async def handle_cat(
         """
         mod_hash, tail_hash, inner_puzzle = curried_args
 
-        hint_list = compute_coin_hints(coin_spend)
-        derivation_record = None
-        for hint in hint_list:
-            derivation_record = await self.puzzle_store.get_derivation_record_for_puzzle_hash(bytes32(hint))
-            if derivation_record is not None:
-                break
+        hinted_coin = hinted_coin_with_coin_id(coin_spend, coin_state.coin.name())
```

Comment:
As I think you've commented and figured out, there's a chance that there's no hint for the coin in here in which case we want to set the derivation record to None and continue on.

Reply:
We don't need to do any cleanup for unhinted coins for now.  We're picking up both hinted and unhinted just fine for a while.  That doesn't matter for this function though.  This function will not be hit except in the case where we do not already have a CAT wallet for an existing CAT in which case a hint is necessary to get our attention.  The segment of code below here does the proper thing, if it sees an unhinted CAT, it likely was not supposed to come here, so it's logging the abnormality as such and then returning None, as you would expect it to do.

Reply:
> That doesn't matter for this function though. This function will not be hit except in the case where we do not already have a CAT wallet for an existing CAT in which case a hint is necessary to get our attention.

Yeah but that case will most likely exist out there and we should take care of it, or do i miss something now? To me it seems like there should be a PR which:

- Adds the missing hinting to the CAT creation code
- Somehow fixes and tests picking up those unhinted coins if there is not yet a wallet 

 The segment of code below here does the proper thing, if it sees an unhinted CAT, it likely was not supposed to come here, so it's logging the abnormality as such and then returning None, as you would expect it to do.

If it sees an unhinted CAT in this place it's definitely an issue, isn't it? Like, a case where we missed to hint the coin.

Reply:
> Adds the missing hinting to the CAT creation code

This PR exists https://github.com/Chia-Network/chia-blockchain/pull/14356, but should not matter for this PR

> Somehow fixes and tests picking up those unhinted coins if there is not yet a wallet

We don't _want_ unhinted coins to trigger wallet creation.  We only want wallet creation to occur upon receiving a hinted coin.  This is how it has always worked (in production, the function you're referring to is only used for tests), so there's no need for backwards compatibility with unhinted stuff.

> If it sees an unhinted CAT in this place it's definitely an issue, isn't it? Like, a case where we missed to hint the coin.

No, exactly not, we only want to pick up hinted coins here.

Reply:
> > Adds the missing hinting to the CAT creation code
> 
> This PR exists #14356, but should not matter for this PR
 
There is fix for the issue in that PR (which still not need to be in there imo and could already be merged independently) yes but nothing which makes sure the coins out in the wild will be picked up properly. Im not sure if either you didn't yet follow what i mean so i will try to express it again:

Imagine you created a CAT in the past which sits in an unhinted coin now, syncing the related wallet will not be able to process the coin because it will just bail out with a "doesn't belong to our wallet" message. How can a coin like this become processed? From my understanding it would never be able to create a wallet. Maybe there is some way to work around this by manually creating the wallet manually or so, i don't know. But this is what i mean with "we hide the actual issue". 

> No, exactly not, we only want to pick up hinted coins here.

And of course, we don't *want* to pick up unhinted coins here.. usually.. but if we had an issue in earlier versions of the code and there actually are valid but unhinted coins like this in existence we can't just ignore them, can we?

And im not necessarily saying we should process or pick up the unhinted coin here in this place although it might be an option to do some checks for it.. but once we come here we already passed `match_cat_puzzle` and we know this coin is a CAT, right? So.. if also the hint is missing, while its supposed to be there if we come here, it means, from my understanding, this case is an issue and should not just silently fail with "not our coin".

Reply:
This is what I'm saying. There was never an issue in an earlier version of the code where _minted_ CATs would not have a hint.  The supported ways to mint have always had a hint.  The function that is failing in the test is a function whose use is limited to tests.  It is not a supported way to mint real CATs.